src/OhioBox.Moranbernate/Querying/QueryExt.cs

```diff
@@ -83,9 +87,11 @@ public static IEnumerable<QueryResult<T>> QueryAggregated<T>(this IDbConnection
 			}
 		}
 
-		private static IEnumerable<T> Run<T>(IDbConnection connection, string sql, List<object> parameters, IEnumerable<Property> properties)
+		private static IEnumerable<T> Run<T>(IDbConnection connection, string sql, List<object> parameters, IEnumerable<Property> properties, Action<SqlDescriptor> trackSql = null)
 			where T : class, new()
 		{
+			trackSql?.Invoke(new SqlDescriptor(sql, parameters));

```

Comment:
extract to method and add try-catch to invocation - since this is an artifact and should not prevent the query to run